server/routes/trial-management/empanel-jury/empanel-jury.controller.js

```diff
@@ -173,20 +182,34 @@
         }));
       }
 
-      delete req.session.trial;
-      delete req.session.trialManagement;
-
       empanelJurorsDAO.post(app, req, {
         jurors,
         trial_number: req.params.trialNumber,
         court_location_code: req.params.locationCode,
         number_requested: requiredNumberOfJurors,
       }).then(() => {
+
+        delete req.session.trial;
+        delete req.session.trialManagement;
+
         return res.redirect(app.namedRoutes.build('trial-management.trials.detail.get', {
           trialNumber: req.params.trialNumber,
           locationCode: req.params.locationCode,
         }));
       }, (err) => {
+        if (err.statusCode === 422) {
+          if (err.error.code === 'JUROR_MUST_BE_CHECKED_IN') {
+            req.session.errors = makeManualError('empanel error', '1 or more jurors have not been checked in today');
+          } else {
+            req.session.errors = makeManualError('empanel error', err.error.message);
+          }
+
+          return res.redirect(app.namedRoutes.build('trial-management.empanel.select.get', {
+            trialNumber: req.params.trialNumber,
+            locationCode: req.params.locationCode,
+          }));
+        }
+
         app.logger.crit('Failed to empanel jurors', {
```

Comment:
Should this log be above the handling, as if we fall into the 422 case we don't get any logging?

Reply:
yes it should